inference-engine/src/preprocessing/ie_preprocess_gapi_kernels.cpp

```diff
@@ -490,6 +561,13 @@ result_t type_dispatch(type_id_t type_id, type_to_id_t&& type_to_id, type_to_val
                                                                      std::forward<default_t>(default_value));
 }
 
+template <typename typelist, typename default_t, typename pred_t, typename type_to_value_t,
+          typename result_t = decltype(std::declval<type_to_value_t>()(type_to_type<head_t<typelist>> {}))>
+result_t type_dispatch(pred_t&& pred, type_to_value_t&& type_to_value, default_t default_value = {}) {
+    return type_dispatch_impl<typelist>::template dispatch<result_t>(std::forward<pred_t>(pred),
+                                                                     std::forward<type_to_value_t>(type_to_value),
+                                                                     std::forward<default_t>(default_value));
+}
```

Comment:
I believe all this template magic should be moved somewhere else (to some header?), even if it is used in this file only.

Reply:
Also, a side note - I find the `splitRow` function earlier in this file looking dramatically long. Is it the _result_ of this refactoring?? Is it applied there already?

Reply:
This refactoring was not applied to split and merge kernels in this PR. splitRowImpl and mergeRowImpl function will be reduced via this approach.